### PR TITLE
feat: Konnect custom DCR docs

### PR DESCRIPTION
**Pull request by c3pko**
### Description

Konnect will now allow customers to create a custom implementation for DCR so they can use an IdP of their choice. We need to point them to the example implementation so they know how to configure it, explain how the configuration works, and also show them the API specs they can use for automation.
 

DOCU-3554
### Testing instructions

Preview link: https://deploy-preview-6662--kongdocs.netlify.app/konnect/dev-portal/applications/dynamic-client-registration/custom/

### Checklist 

- [x] Review label added 
- [x] PR pointed to correct branch (`main` for immediate publishing, or a release branch: e.g. `release/gateway-3.2`, `release/deck-1.17`)

### CODE REVIEW

File: app/konnect/dev-portal/applications/dynamic-client-registration/custom.md
Comment (cloudjumpercat) [inline app/konnect/dev-portal/applications/dynamic-client-registration/custom.md:15]: @c3pko "This integration can act as either a translation layer to bridge to your network"

With the "either" in there, I think this sentence was meant to continue? What else can the integration act as other than the translation layer?

File: app/konnect/dev-portal/applications/dynamic-client-registration/custom.md
Comment (cloudjumpercat) [inline app/konnect/dev-portal/applications/dynamic-client-registration/custom.md:31]: @c3pko Just to clarify that I'm understanding how this works, how is the developer "shown" the credentials? Is the CDC part of the IdP? I'm not sure where the IdP fits in the diagram.

File: app/konnect/dev-portal/applications/dynamic-client-registration/custom.md
Comment (cloudjumpercat) [inline app/konnect/dev-portal/applications/dynamic-client-registration/custom.md:35]: @c3pko What do you integrate the DCR handler with? Your IdP? Konnect?

File: app/konnect/dev-portal/applications/dynamic-client-registration/custom.md
Comment (cloudjumpercat) [inline app/konnect/dev-portal/applications/dynamic-client-registration/custom.md:37]: @c3pko Does "operation" here mean what the CDC passes to the translation layer?

File: app/konnect/dev-portal/applications/dynamic-client-registration/custom.md
Comment (nateslo) [inline app/konnect/dev-portal/applications/dynamic-client-registration/custom.md:15]:
```suggestion
If your third-party IdP is not on this list, you can still use your IdP with {{site.konnect_short_name}} by using a custom HTTP DCR bridge. This HTTP DCR bridge acts as a proxy and translation layer between your IdP and DCR applications in the Dev Portal. When a developer creates a DCR application in the Dev Portal, Konnect calls your HTTP DCR bridge which can translate the application data into a format suitable for your third-party IdP. 
```

File: app/konnect/dev-portal/applications/dynamic-client-registration/custom.md
Comment (nateslo) [inline app/konnect/dev-portal/applications/dynamic-client-registration/custom.md:35]:
```suggestion
To use an unsupported IdP with DCR, you must implement the [{{site.konnect_short_name}} Dev Portal DCR Handler spec](https://github.com/Kong/konnect-portal-dcr-handler/blob/main/openapi/openapi.yaml). Kong provides an example reference implementation in the [{{site.konnect_short_name}} Dev Portal DCR Handler repository](https://github.com/Kong/konnect-portal-dcr-handler). This is an example HTTP DCR bridge implementation and is not meant to be deployed in production. We encourage you to use this implementation as a guide to create your own implementation. 
```

File: app/konnect/dev-portal/applications/dynamic-client-registration/custom.md
Comment (nateslo) [reply app/konnect/dev-portal/applications/dynamic-client-registration/custom.md:37]: Operation means an HTTP request in this context

Comment (nateslo) [inline app/konnect/dev-portal/applications/dynamic-client-registration/custom.md:37]:
```suggestion
Any request that does not return a `2xx` status code is considered failing and won't continue the process of trying to create an application in your {{site.konnect_short_name}} portal.
```

**Comment by cloudjumpercat** [comment]
@c3pko @nateslo Thanks for the feedback! I just did some of my own copyedits, so I think the only thing I have questions about is if the custom DCR spec will be a new spec or an addition to an existing spec? This will help me know what additional things I should add to this PR depending on if it's new or just an addition to existing.

**Comment by nateslo** [comment]
@cloudjumpercat The spec currently lives in the example repo here: https://github.com/Kong/konnect-portal-dcr-handler/blob/main/openapi/openapi.yaml. 

I don't think we want to host the spec in docs because we don't actually have an API that backs the spec. Customers need to implement an API to conforms to this spec to use the HTTP DCR feature

File: app/konnect/dev-portal/applications/dynamic-client-registration/custom.md
Comment (cloudjumpercat) [inline app/konnect/dev-portal/applications/dynamic-client-registration/custom.md:35]:
```suggestion
To use an unsupported IdP with DCR, you must implement an API that conforms to the [{{site.konnect_short_name}} Dev Portal DCR Handler spec](https://github.com/Kong/konnect-portal-dcr-handler/blob/main/openapi/openapi.yaml). Kong provides an example reference implementation in the [{{site.konnect_short_name}} Dev Portal DCR Handler repository](https://github.com/Kong/konnect-portal-dcr-handler). This is an example HTTP DCR bridge implementation and is not meant to be deployed in production. We encourage you to use this implementation as a guide to create your own implementation. 
```

**Review by cloudjumpercat — APPROVED** [review]
LGTM! Thanks for making these changes and for the feedback! This can be merged once the feature goes GA (docs will be published immediately once the PR is merged).

**Comment by nateslo** [comment]
> LGTM! Thanks for making these changes and for the feedback! This can be merged once the feature goes GA (docs will be published immediately once the PR is merged).

Thank you @cloudjumpercat!

**Pull request merged**
